GitCommands/Git/GitCommandHelpers.cs

```diff
@@ -608,14 +608,40 @@ public static bool Plink()
             return sshString.EndsWith("plink.exe", StringComparison.CurrentCultureIgnoreCase);
         }
 
-        /// <summary>Gets the git SSH command; or "" if the environment variable is NOT set.</summary>
+        /// <summary>
+        /// Gets the git SSH command;
+        /// If the environment variable is not set, will try to find ssh.exe in git installation directory;
+        /// If not found, will return "".
+        /// </summary>
         public static string GetSsh()
         {
             var ssh = Environment.GetEnvironmentVariable("GIT_SSH", EnvironmentVariableTarget.Process);
 
+            if (string.IsNullOrEmpty(ssh))
+            {
+                ssh = GetSshFromGitDir();
+            }
+
             return ssh ?? "";
         }
 
+        private static string GetSshFromGitDir()
+        {
+            var gitBinDir = AppSettings.GitBinDir;
+            if (string.IsNullOrEmpty(gitBinDir))
+            {
+                return null;
+            }
+            var gitBinDirInfo = new DirectoryInfo(gitBinDir);
+            var gitDirInfo = gitBinDirInfo.Parent;
+            if (gitDirInfo == null)
+            {
+                return null;
+            }
+            var sshExe = gitDirInfo.GetFiles("ssh.exe", SearchOption.AllDirectories).FirstOrDefault();
```

Comment:
according to the MS docs this is not the most efficient way - [see Remarks section](https://msdn.microsoft.com/en-us/library/dd413232(v=vs.110).aspx?f=255&mspperror=-2147217396#Anchor_2):

> The EnumerateFiles and GetFiles methods differ as follows:
> * When you use EnumerateFiles, you can start enumerating the collection of FileInfo objects before the whole collection is returned.
> * When you use GetFiles, you must wait for the whole array of FileInfo objects to be returned before you can access the array.
>
> Therefore, when you are working with many files and directories, EnumerateFiles can be more efficient.

On top of that [see Examples section](https://msdn.microsoft.com/en-us/library/dd413232(v=vs.110).aspx?f=255&mspperror=-2147217396#Anchor_3):

> If you only need the names of the files, use the static Directory class for better performance. For an example, see the Directory.EnumerateFiles(String) method.